### ISSUE TITLE
'AE' object has no attribute 'torch'plate

### ISSUE THREAD

**Issue by jdeawick** [mrdbourke/pytorch-deep-learning#462]
### Discussed in https://github.com/mrdbourke/pytorch-deep-learning/discussions/117

<div type='discussions-op-text'>

<sup>Originally posted by **mrdbourke** August 31, 2022</sup>
This template is to help with the formatting of questions to best help others who are reading it.

1) "Learn pyTorch for deep learning in a day. Literally" [timestamp 6:49:00]

2) Error  'self.L1' cannot contain assignment
    TODO: Error occurs in the looping process (i.e., 'L1 is not recognized)

3) Code: 
\\\# AutoEncoder for unsupervised classification
\\\import arcgis, arcpy, os, string, time
\\\from arcpy import env
\\\from arcpy.sa import *
\\\arcpy.CheckOutExtension("spatial")
\\\import pandas as pd
\\\import numpy as np
\\\import sklearn as sk
\\\from sklearn import model_selection
\\\import torch 
\\\from torch import nn, utils, backends
\\\from torch.utils import data
\\\from torch.utils.data import DataLoader, Dataset
\\\import import collections 
\\\from collections import OrderedDict
\\\# 
\\\print(time.strftime('%#c',time.localtime()))
\\\env.workspace = r'E:\PFL2019\PFLdata2019.gdb'
\\\env.outputCoordinateSystem = 'state48west'
\\\env.parallelProcessingFactor = '100%'
\\\#
\\\print(torch.__version__) -> 1.8.2 (old, I know)
\\\x = pd.read_csv(r'E:\PFL2019\SASinputFiles\GausMixO2.csv', header=0, index_col=0)
\\\x = x[['VI01', 'VI02', 'VI03', 'VI04', 'VI05', 'VI06', 'VI07', 'VI08', 'VI09','VI10', 'VI11', 'VI12', 'VI13', 'VI14', 'VI15', 'VI16', 'VI17', 'VI18','VI19', 'VI20', 'VI21', 'VI22']]
\\\xray=x.to_numpy()
\\\tx = torch.from_numpy(xray)
\\\txr = tx.float()
\\\class AE(nn.Module.children()):
\\\  def __init__(self):
\\\    super().__init__()
\\\    nn.Sequential(
\\\      self.L1 = torch.nn.Linear(in_features=22, out_features=22, bias=True),
\\\      self.E1 = torch.nn.ELU(alpha=8),
\\\      self.L2 = torch.nn.Linear(22, 20, bias=True),
\\\      self.E2 = torch.nn.ELU(alpha=8),
\\\      self.L3 = torch.nn.Linear(20, 18, bias=True),
\\\      self.E3 = torch.nn.ELU(alpha=8),
\\\      self.L4 = torch.nn.Linear(18, 16, bias=True),
\\\      self.E4 = torch.nn.ELU(alpha=8),
\\\      self.L5 = torch.nn.Linear(16, 14, bias=True),
\\\      self.E5 = torch.nn.ELU(alpha=8),
\\\      self.L6 = torch.nn.Linear(14, 12, bias=True),
\\\      self.E6 = torch.nn.ELU(alpha=8),
\\\      self.L7 = torch.nn.Linear(12, 10, bias=True),
\\\      self.E7 = torch.nn.ELU(alpha=8),
\\\      self.L8 = torch.nn.Linear(10, 8, bias=True),
\\\      self.E8 = torch.nn.ELU(alpha=8),
\\\      self.L9 = torch.nn.Linear(8, 6, bias=True),
\\\      self.E9 = torch.nn.ELU(alpha=8)
\\\    )
\\\    def forward(self, x: torch.Tensor):
\\\      x = self.E1(self.L1(x))
\\\      x = self.E2(self.L2(x))
\\\      x = self.E3(self.L3(x))
\\\      x = self.E4(self.L4(x))
\\\      x = self.E5(self.L5(x))
\\\      x = self.E6(self.L6(x))
\\\      x = self.E7(self.L7(x))
\\\      x = self.E8(self.L8(x))
\\\      x = self.E9(self.L9(x))
\\\      return x
\\\ ERROR: "self.L1" = ...
\\\ ERROR: expression cannot contain assignment, perhaps you meant ==?
\\\ COMMENT:  self.x is in every example I've seen on the internet???
\\\ m=AE()
\\\ Lossfx = torch.nn.MSELoss()
\\\ opt=torch.optim.SGD(params=m.parameters(), lr=0.01, momentum=0.9)
\\\ epochs=10
\\\ for epoch in range(epochs):
\\\   m.train()
\\\   pred = m(txr)
\\\   loss=Lossfx(pred,m(txr)).backward()
\\\   print(f'loss: {loss}')
\\\   opt.zero_grad()
\\\   opt.step()
\\\ pred[0]
\\\ ### unsupervised classification
\\\ ypred=torch.nn.Softmax(dim=1(pred)

4) Troubleshooting
    a) tried adding the suffixes:  def __int__(self, L1, ..., E9)
    b) tried add_module
        super().__init__()
        self.L1.add_module('L1',torch.nn.Linear(22,22,bias=True))
        ...
        self.E9. ....
        self = nn.Sequential(
    c) used an ordered dictionary inside nn.Sequential
        self = nn.Sequential(OrderedDict([ 
        ('L1',torch.nn.linear(22,22,bias=True)),
        ...
        ('E9', torch.nn.ELU))
        ]))
    Comment: a) returned a positional error argument; b) gave the same error as described above; c) completed the class definition, 
    but no parameters were returned after initiation (m=AE()) **list(m.parameters() returned []*** and ** pred=m(txr) returned 'AE has 
    no object E1**
    I'm stumped!
